src/compose/__tests__/graph.test.ts

```diff
@@ -0,0 +1,187 @@
+import { literal, optional } from "@spot"
+import { bind, createTag } from "@tag"
+import { createTask } from "@task"
+import { describe, expect, it } from "vitest"
+import { graph } from "../graph"
+import type { Stage } from "../types"
+
+const alphaTask = createTask({ name: "alpha", run: { fn: () => ({ value: true }) } })
+
+describe("graph tests", () => {
+  it("zero dependencies", () => {
+    const stages: [Stage] = [[alphaTask]]
+    const expected = [{ id: 0, name: "Task[alpha]", type: "task", dependencies: { optional: [], required: [] } }]
+
+    expect(graph(stages)).toStrictEqual(expected)
+  })
+
+  it("direct dependency on another task", () => {
+    type BetaCtx = { value: boolean }
+    const betaTask = createTask({
+      name: "beta",
+      run: { fn: (ctx: BetaCtx) => !ctx.value, context: { value: alphaTask.value } },
+    })
```

Comment:
nit: to simplify we can set the context to boolean directly
```suggestion
    const betaTask = createTask({
      name: "beta",
      run: { fn: (ctx: boolean) => !ctx.value, context: alphaTask.value },
    })
```